### PR TITLE
add Redux reducers and actions

### PR DESCRIPTION
**Pull request by Kakalanp**
Adding Action, Action-types and reducers for categories and books, including helper functions for the last one.

### CODE REVIEW

File: src/actionTypes.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
+const CHECK = 'CHECK';
```
Comment (Mihndim2020) [inline src/actionTypes.js]: - [x] Great job setting up the action types, but there is an issue we need to fix. According to the project requirements, which is a best practice, we are supposed to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), something like so(`just example`); `const ADD_BOOK = bookstore/books/ADD_BOOK`, `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK`, `const CHECK_STATUS = bookstore/categories/CHECK_STATUS`.  Please let's fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @Kakalanp, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Your project is running free of `Linter` errors.
:heavy_check_mark: Descriptive Readme file.

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/actionTypes.js (hunk 1 above)
Comment (Kakalanp) [reply src/actionTypes.js]: I missed the requirement. Thanks for pointing it out! 😉

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Kakalanp, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kakalanp** [comment]
Thank you very much @Mihndim2020!

**Pull request merged**